components/dot_menu/dot_menu.jsx

```diff
@@ -116,124 +105,126 @@ export default class DotMenu extends Component {
     render() {
         const isSystemMessage = PostUtils.isSystemMessage(this.props.post);
         const isMobile = Utils.isMobile();
+        const canDelete = PostUtils.canDeletePost(this.props.post);
+        const canEdit = PostUtils.canEditPost(this.props.post, this.editDisableAction); // Fix this crazy
```

Comment:
This function call set's a timer that calls back to the editDisableAction in the case where the post needs to be non-editable. This logic should be top down, not here. Hence my comment.